src/App.js

```diff
@@ -1,16 +1,43 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatEntry from './components/ChatEntry';
+import ChatLog from './components/ChatLog';
+
 
 const App = () => {
+
+  const [chatData, setChatData] = useState(chatMessages);
+  let [likeCount, setLikeCount] = useState(0);
+  // let likeCount = 0;
+
+  const onLikeClick = (id) => {
+  setChatData(chatData.map(chat => {
+    if (chat.id === id){
+      chat.liked = !chat.liked
+      if (chat.liked === true){
+        setLikeCount(likeCount += 1)
+      };
+      if (chat.liked === false){
+        setLikeCount(likeCount -= 1)
+      };
+      
+      return chat;
+    }
+    return chat;
+}));
+};
```

Comment:
The way `likeEntry` updates the messages data is what's causing one of the tests to fail depending on the order it runs in. 
> "In react-chatlog, inadvertently mutating the state object leads to inadvertently mutating the [module-static ](https://github.com/tetianazakipna/react-chatlog/blob/main/src/App.js#L3)messages[ variable](https://github.com/tetianazakipna/react-chatlog/blob/main/src/App.js#L3), which has the side effect of persisting “state” across the two test runs.

Our useState data `chatData` is intended to be read-only, but we're editing the existing entries on line 15. What we want to do is make a copy of the message object we're updating, change the liked value on the copy, then return that new object. Essentially, we can make a shallow copy of objects that are not changing, but we need a deep copy of any objects we're updating. One way we could do that might look like:
```javascript
  const onLikeClick = (id) => {
    const newEntries = chatData.map((chat) => {
      if (chat.id === id) {
        // Use spread operator to copy all of entry's values to a new object, 
        // then overwrite the attribute liked with the opposite of what entry's liked value was
        return {...chat, liked: !chat.liked}
      } 

      // We didn't enter the if-statement, so this entry is not changing. 
      // Return the existing data
      return chat;
    });
    
    setChatData(newEntries);
  };
```